### ISSUE TITLE
Jobspec: specification of jobshell attributes?

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/rfc#269]
In a recent Exaworks call, @andre-merzky pointed out that we don't have the specification of stdout/stderr file redirection documented.  To some extent, I think we intentionally aren't specifying jobshell options in RFC14, but does it make sense for common options like stdout/stderr redirection?

**Comment by grondo** [comment]
Yeah, we left job-shell options out of the RFC since the job-shell could technically be replaced (and also the input and output plugins could be replaced). However, maybe allowing that level of customization is just asking for trouble? Plus it is doubtful anyone would replace these builtin plugins.

It might also be sufficient if we could make the shell options provide by plugins self-documenting.  E.g. Slurm plugins can add new longopts to `srun`, `salloc`, `sbatch` that appear in `--help` output along with usage descriptions. I wonder if we could enable something like that for shell plugins which read shell options.
